Makefile

```
@@ -123,11 +123,40 @@ export KOKKOS_DEPS := $(KOKKOS_LIB)
 export KOKKOS_CXXFLAGS := -I$(KOKKOS_INSTALL)/include
 $(eval $(call CUFLAGS_template,$(KOKKOS_CUDA_ARCH),KOKKOS_))
 KOKKOS_CUDA_CUFLAGS := $(KOKKOS_NVCC_COMMON) $(USER_CUDAFLAGS)
-export KOKKOS_CUFLAGS := $(KOKKOS_CUDA_CUFLAGS) -Xcudafe --diag_suppress=esa_on_defaulted_function_ignored
+export KOKKOS_CUFLAGS := $(KOKKOS_CUDA_CUFLAGS)
 export KOKKOS_LDFLAGS := -L$(KOKKOS_INSTALL)/lib -lkokkoscore -ldl
 export KOKKOS_DLINKFLAGS := $(KOKKOS_CUDA_DLINKFLAGS)
 export NVCC_WRAPPER_DEFAULT_COMPILER := $(CXX)
 
+# SYCL / oneAPI and compatibility tool
+SYCL_BASE  := /opt/intel/oneapi/compiler/latest/linux
+DPCT_BASE  := /opt/intel/oneapi/dpcpp-ct/latest
+ONEAPI_ENV := /opt/intel/oneapi/setvars.sh
```

How about
```suggestion
ONEAPI_BASE := /opt/intel/oneapi
SYCL_BASE  := $(ONEAPI_BASE)/compiler/latest/linux
DPCT_BASE  := $(ONEAPI_BASE)/dpcpp-ct/latest
ONEAPI_ENV := $(ONEAPI_BASE)/setvars.sh
```
? Then it would be sufficient to override just `ONEAPI_BASE` for a different location of oneAPI distribution (while still preserving the ability to override individual variables)

---

The reasons is an attempt of compatibility with the open source llvm/clang/sycl compiler, which does not have `dpct` and `setvars.sh`.

For `dpct` I take it from the latest oneAPI beta, so it need a potentially different `BASE` directory.
For `setvars.sh` just do without.

Let me see if I can play with the variables to use a common default, at least.

---

ok, it's probably not worth doing anything extra complicated right now... I'll have to revisit this anyway next time I try a different compiler.